templates/.github/workflows/go-test.yml

```diff
@@ -11,7 +11,7 @@ jobs:
     env:
       COVERAGES: ""
       SKIP32BIT: false
-    runs-on: ${{ format('{0}-latest', matrix.os) }}
+    runs-on: ${{ fromJSON(vars[format('UCI_GO_TEST_RUNNER_{0}', matrix.os)] || format('"{0}-latest"', matrix.os)) }}
```

Comment:
> > Even though https://github.com/orgs/community/discussions/44322 hasn't been picked up by anyone from GitHub yet, I think we should proceed with this solution. It introduces a way for experimentation with self-hosted runners. In this context, it could be viewed as an upside that it wouldn't extend to external contributors' PRs yet
> 
> @galargh Can you clarify what happens when run from a fork? It would claim that it's a requirement that PRs authored by outside contributors (i.e. on their forks) work, otherwise we won't be able to accept any community contributions.

Sure! PRs from forks do not have access to `vars` yet, so, even if the custom runner variable is set, they'll be using a fallback to `format('"{0}-latest"', matrix.os)` (the default hosted runners).